src/import/FSHImporter.ts

```diff
@@ -1245,6 +1250,18 @@ export class FSHImporter extends FSHVisitor {
     return reference.slice(reference.indexOf('(') + 1, reference.length - 1).split(/\s*\|\s*/);
   }
 
+  visitCanonical(ctx: pc.CanonicalContext): FshCanonical {
+    const canonicalText = this.parseCanonical(ctx.CANONICAL().getText());
+    const fshCanonical = new FshCanonical(canonicalText)
+      .withLocation(this.extractStartStop(ctx))
+      .withFile(this.currentFile);
+    return fshCanonical;
+  }
+
+  private parseCanonical(canonical: string): string {
+    return canonical.slice(canonical.indexOf('(') + 1, canonical.length - 1);
```

Comment:
The grammar allows spaces inside (like `Canonical(    Foo     )`).  Do we need to trim those?  It looks like we _don't_ trim them; I just can't tell if that's because we don't need to or if it's a mistake.

Reply:
OK.  I checkout out the branch and updated the import test to have spaces inside the `(` `)`.  The result is that it created a `FshCanonical` whose entity name had spaces in it.  I then updated an exporter test to see what happens if a `FshCanonical` has spaces in the entity name, and it failed.  So... I think we need to trim the spaces!

Reply:
Ah I think I was using `Reference` as a guide when writing the grammar, which also allows spaces. You're definitely right that we should trim the spaces so we can identify the entity, but I think I also just want to double check that we do in fact want the spaces included in the grammar, and that I didn't make a bad assumption there.

Reply:
I think the spaces are good.  I'd say keep them in the grammar -- let's just make sure we handle them appropriately...

I wonder if maybe we don't handle `Reference`s with spaces appropriately then?  It may be worth checking.

Reply:
I had the same thought and looked into it, and I don't think it's being handled correctly. If you have the following rule in your FSH

```
* subject = Reference(   mCODEPatientExample01 )
```
the output will be 
```
"reference": "   mCODEPatientExample01 "
```
which is probably not what was intended, but no errors are logged.

Additionally, if you have the following rule in  your FSH
```
* stage.assessment only Reference(   CancerStageParent )
``` 
SUSHI logs the following error and skips the rule:
```
error No definition for the type "   CancerStageParent " could be found.
```

Reply:
Yep.  Then that is broken for sure.  Can you maybe file a new bug for that and just address `Canonical` in this PR?

Reply:
Sure! I'll log an issue for References.